lib/datasets.go

```diff
@@ -248,13 +201,25 @@ func scriptFileSelection(ds *dataset.Dataset, selector string) (qfs.File, bool)
 
 // ActivityParams defines parameters for the Activity method
 type ActivityParams struct {
-	ListParams
+	params.List `json:"list,omitempty"`
+	// Term filters for "history" or "run" activity type
+	Term string `json:"term,omitempty"`
```

Comment:
Filtering is pretty up in the air right now. It seems to only really be used in two cases:
1) in search
2) in Activity (between "history" & "run")

Side stepping this problem for now, by just not using `Filter` and instead staying with using `Q` field in `SearchParams` and `Term` field in `ActivityParams`.

However, one potential way to use the filter field in `SearchParams`:
`/search?filter=q:my_search_term`

`ActivityParams`:
`/log?filter=type:history`
`/log?filter=type:run`

Reply:
I am fine with putting something temporarily in place, as I agree it isn't well defined. `Term` looks good for now!